src/lib.rs

```diff
@@ -39,25 +47,38 @@ pub struct SphinxPacket {
 
 impl SphinxPacket {
     pub fn new(
-        message: Vec<u8>,
+        mut message: Vec<u8>,
         route: &[Node],
         destination: &Destination,
         delays: &[Delay],
```

Comment:
It's more of a future note for refactoring: if we're already encapsulating route, delays and destination for surb_materials, maybe let's do the same thing for main_route_materials?  (i.e. encapsulate current `route`, ` destination` and `delays` in something)